### PR TITLE
Add support for passing keyword args to `data` in JS client

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

This PR adds support for passing keyword arguments as well as positional args to the `data` param in the JS client. This is backwards compatible, so the two usages will work:

```
const result = await app.predict("/predict", ["Chewbacca"]);
```
```
const result = await app.predict("/predict", { name: "Chewbacca" });
```
Also adds unit tests. 

The `CodeSnippet.svelte` API usage has been updated, which you can see here: https://huggingface.co/spaces/hmb/test_args

Closes: #7831

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: client/js/src/helpers/api_info.ts [hunk 1]
```diff
@@ -298,3 +298,72 @@ export function handle_message(
...
+	const resolved_data: unknown[] = [];
+	const provided_keys = Object.keys(data);
+
+	parameters.forEach((param, index) => {
+		if (data.hasOwnProperty(param.parameter_name)) {
+			resolved_data[index] = data[param.parameter_name];
+		} else if (param.parameter_has_default) {
+			resolved_data[index] = param.parameter_default;
+		} else {
+			resolved_data[index] = undefined;
```
Comment (pngwn) [inline client/js/src/helpers/api_info.ts:348]: Should we throw an error in this case? 

@abidlabs @freddyaboulton what does the python client do if no param is given but there is no default?

**Review by pngwn — APPROVED** [review]
Just left one small comment but everything else looks great + works as expected!

Thanks @hannahblair!

File: client/js/src/helpers/api_info.ts (hunk 1 above)
Comment (abidlabs) [reply client/js/src/helpers/api_info.ts:348]: Yes it raises an error saying that a required param is missing

**Comment by abidlabs** [comment]
Nice @hannahblair! Just 2 nits with regards to the code snippets:

1)

image


* The code snippet itself looks good, but in the parameter table below, we should now put the name of the parameter, instead of the `[0]` to designate the positional argument
* We can also remove the comment `// string in...` as that information is included in the parameter table

Matching the Python code snippet and parameter table:

image

2) The RecordingSnippet.svelte code snippet still shows the positional format, would be great if we can update this as well:

image

**Review by pngwn — APPROVED** [review]
Tested and looks good, will file another issue for the API recorder because it is broken in the release.

**Pull request merged**